File: #0078.subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {

        int size = nums.size();
        vector<int> work;
        vector<vector<int>> result;    
        for (int i = 0; i < 1 << size; ++i) {
            work.clear();
            for (int j = 0; j < size; ++j) {
                if (i & (1 << j)) {
                    work.push_back(nums[j]);
                }
            }
            result.push_back(work);
        }
        return result;
    }
};

File: #0078.subsets.rs
impl Solution {
    pub fn subsets(nums: Vec<i32>) -> Vec<Vec<i32>> {

        let mut result = vec![vec![]];
        let mut work = vec![vec![]];
        for num in nums {
            work = result.clone();
            work.iter_mut().for_each(|x| x.push(num));
            result.append(&mut work);
        }
        return result;
    }
}
